src/NuGet.Clients/PackageManagement.VisualStudio/ProjectSystems/LegacyCSProjPackageReferenceProject.cs

```diff
@@ -278,26 +281,35 @@ private PackageSpec GetPackageSpec()
 
             // In legacy CSProj, we only have one target framework per project
             var tfis = new TargetFrameworkInformation[] { projectTfi };
-            return new PackageSpec(tfis)
+
+            return new PackageSpec()
             {
                 Name = _projectName ?? _projectUniqueName,
+                Version = new NuGetVersion(1, 0, 0, 0, (string)null, string.Empty),
+                Authors = new string[] { },
+                Owners = new string[] { },
+                Tags = new string[] { },
+                TargetFrameworks = tfis,
+                ContentFiles = new string[] { },
+                Dependencies = packageReferences,
+                IsDefaultVersion = true,
                 FilePath = _projectFullPath,
                 RuntimeGraph = runtimeGraph,
                 RestoreMetadata = new ProjectRestoreMetadata
                 {
-                    OutputType = RestoreOutputType.NETCore,
+                    OutputType = RestoreOutputType.UAP,
                     OutputPath = GetBaseIntermediatePath(),
                     ProjectPath = _projectFullPath,
                     ProjectName = _projectName ?? _projectUniqueName,
                     ProjectUniqueName = _projectFullPath,
                     OriginalTargetFrameworks = tfis
-                        .Select(tfi => tfi.FrameworkName.GetShortFolderName())
+                        .Select(tfi => tfi.FrameworkName?.GetShortFolderName())
```

Comment:
This should never be null, right?
Also, if it were null you would need to filters those out